virtcontainers/kata_agent.go

```diff
@@ -505,9 +507,27 @@ func (k *kataAgent) startSandbox(sandbox *Sandbox) error {
 		Options:    sharedDir9pOptions,
 	}
 
+	storages := []*grpc.Storage{sharedVolume}
+
+	if sandbox.shmSize > 0 {
+		path := filepath.Join(kataGuestSharedDir, shmDir)
+		shmSizeOption := fmt.Sprintf("size=%d", sandbox.shmSize)
+
+		shmStorage := &grpc.Storage{
+			Driver:     kataEphemeralDevType,
+			MountPoint: path,
+			Source:     "shm",
```

Comment:
@bergwolf I just realised that the `source` in case of /dev/shm is set to `shm` and not `tmpfs`, various for ephemeral volumes it is `tmpfs` .
```
$ sudo docker run -it --runtime=runc  debian sh -c "mount | grep shm"
shm on /dev/shm type tmpfs (rw,nosuid,nodev,noexec,relatime,size=65536k)
```
This source field would need to be changed to `tmpfs` to use the ephemeral storage. I know shm is a special case of tmpfs, but I havent looked into the implications of doing this,  do you think it is ok to use `tmpfs` instead of `shm`.

I would prefer to introduce a new storage type for shm to be consistent with runc behaviour.

Reply:
@amshinde for ephemeral storage, the source actually does not matter. You can set it as `shm` or `tmpfs` or whatever, and it will be shown as such inside the container. Do you mean that kata-agent would only accept `tmpfs` source for ephemeral storage? If so, we need to fix it in kata-agent side.

Reply:
@bergwolf Yes, you are right. I set this to shm and tested it.